Libraries/LibTLS/TLSv12.cpp

```diff
@@ -137,7 +138,22 @@ ErrorOr<bool> TLSv12::can_read_without_blocking(int timeout) const
     if (!m_ssl)
         return Error::from_string_literal("SSL connection is closed");
 
-    return m_socket->can_read_without_blocking(timeout);
+    if (SSL_has_pending(m_ssl))
+        return true;
+
+    while (timeout > 0) {
+        auto timer = Core::ElapsedTimer();
+        if (!TRY(m_socket->can_read_without_blocking(timeout)))
+            return SSL_has_pending(m_ssl);
+        if (SSL_has_pending(m_ssl))
+            return true;
+        auto elapsed = timer.elapsed_milliseconds();
+        if (elapsed >= timeout)
+            break;
+        timeout -= elapsed;
+    }
```

Comment:
This is a kinda confusing way to write this. What's the units of `timeout` ? milliseconds?

Could you not create one ElapsedTimer at the top before the loop and check that it hasn't "expired" ? And pass remaining_ms() or similar to the underlying socket's can_read_without_blocking?

Presumably the loop is in place to account for spurious wakeups from the socket's select/poll call? and/or wakeups that don't result in a full-enough set of data from OpenSSL?

Reply:
Is there a way to check for "expired"? I mean sure, we can make the timer outside the loop and pass timeout-elpased I guess